### PR TITLE
feat(web3): add send transaction error with logs #18277

### PR DESCRIPTION
**Pull request by beautyfree**

### CODE REVIEW

File: web3.js/src/errors.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export class SendTransactionError extends Error {
+  logs: string | undefined;
+
+  constructor(message: string, logs?: string) {
```
Comment (jstarry) [inline web3.js/src/errors.ts:4]: logs will either be undefined or an array of strings. Can you please update the type of `logs` everywhere to indicate that it's an array?

Comment (beautyfree) [reply web3.js/src/errors.ts]: oh yeah, of course

**Comment by jstarry** [comment]
Could you please fix the prettier check too?

**Review by jstarry — APPROVED** [review]

**Comment by jstarry** [comment]
Thanks for your contribution @beautyfree!

**Pull request merged**